### PR TITLE
Adding unpack-characterizations CLI

### PR DESCRIPTION
**Pull request by mjgleason**
This PR introduces the `unpack-characterizations` CLI. This command can be used to batch unpack the various "characterization" columns often included in reV supply curves. These columns typically have values that are JSON strings, and the purpose of this tool is to explode those JSON strings into new columns, optionally converting the values to area in units of square kilometers. The output from the tool is a copy of the supply curve with the new columns appended to the end. This tool is designed to be very performant and is able to unpack several characterization columns from a full size CONUS supply curve in a matter of seconds. 

One of the inputs to the command is a "characterization map" which is a JSON file that specifies which columns to unpack and how to unpack them. There are details on this input in the documentation of `reView.utils.characterization.unpack_characterizations()` and an example JSON file can be found at `tests/data/characterization-map.json`.

This PR is draft because more I have only implemented very basic testing so far. Sharing it now in case (a) it's useful in the near term and (b) you want to review it sooner than later.

### CODE REVIEW

**Comment by ppinchuk** [comment]
Cool, will keep an eye on this! First pass glance looks great to me. Obv some stuff to finish like docstrings and tests, but the code logic looks sound

File: reView/cli.py [hunk 1]
```diff
@@ -83,3 +83,49 @@ def unpack_turbines(
...
+              type=click.Path(exists=True, dir_okay=False, file_okay=True),
+              help='Path to JSON file storing characterization map')
+@click.option('--out_csv', '-o', required=True,
+              prompt='Path to output csv.',
+              type=click.Path(dir_okay=False),
+              help='Path to CSV to store results')
+@click.option('--cell_size', '-c', required=False,
+              default=90.,
+              type=float,
+              help=('(Optional) Cell size in meters of characterization '
```
Comment (ppinchuk) [inline reView/cli.py:104]: Doesn't `click` tell the use automatically if a parameter is optional? Also you don't list the following parameter as `(Optional)`, which is slightly inconsistent.

File: reView/utils/characterizations.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    Returns
+    -------
+    pandas.DataFrame
+        New pandas dataframe with additional columns for data unpacked
+        from characterization columns.
+
+    Raises
+    ------
+    ValueError
+        _description_
```
Comment (ppinchuk) [inline reView/utils/characterizations.py:148]: Missing info here, I think?

**Review by ppinchuk — APPROVED** [review]
Nice and thorough work! Great job on the tests + docs in particular. Just a few minor comments from me, but overall, I think this is excellent utility for reView

**Comment by WilliamsTravis** [comment]
This sounds excellent, I will review this soon.

**Review by WilliamsTravis — APPROVED** [review]
Looks good to me.

**Pull request merged**